src/bridging/providers/across/AcrossApi.ts

```diff
@@ -0,0 +1,371 @@
+  }: AvailableRoutesRequest): Promise<Route[]> {
+    const params: Record<string, string> = {}
+    if (originChainId) params.originChainId = originChainId
+    if (originToken) params.originToken = originToken
+    if (destinationChainId) params.destinationChainId = destinationChainId
+    if (destinationToken) params.destinationToken = destinationToken
+
+    return this.fetchApi('/available-routes', params, isValidRoutes)
+  }
+
+  /**
+   * Retrieve suggested fee quote for a deposit.
+   *
+   * Returns suggested fees based inputToken+outputToken, originChainId, destinationChainId, and amount.
+   * Also includes data used to compute the fees.
+   *
+   * * See https://docs.across.to/reference/api-reference#suggested-fees
+   */
+  async getSuggestedFees(request: SuggestedFeesRequest): Promise<SuggestedFeesResponse> {
+    const params: Record<string, string> = {
+      token: request.token,
+      originChainId: request.originChainId.toString(),
+      destinationChainId: request.destinationChainId.toString(),
+      amount: request.amount,
+    }
+
+    if (request.recipient) {
+      params.recipient = request.recipient
+    }
+
+    // Get the quote from the Across API (see https://docs.across.to/reference/api-reference#suggested-fees)
+    // Example: https://app.across.to/api/suggested-fees?token=0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913&originChainId=8453&destinationChainId=137&amount=100000000
+    //
+    // TODO: The API documented params don't match with the example above. Ideally I would use 'inputToken' and 'outputToken', but the example above uses 'token'. This will work for current implementation, since we bridge the canonical token, but this will need to be reviewed
+    //       https://app.across.to/api/suggested-fees?inputToken=0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913&originChainId=8453&destinationChainId=137&outputToken=0xc2132D05D31c914a87C6611C10748AEb04B58e8F&amount=100000000
+    const fees = await this.fetchApi('/suggested-fees', params, isValidSuggestedFeesResponse)
+
+    return fees
+  }
+
+  protected async fetchApi<T>(
+    path: string,
+    params: Record<string, string>,
+    isValidResponse?: (response: unknown) => response is T
+  ): Promise<T> {
+    const baseUrl = this.options.apiBaseUrl || ACROSS_API_URL
+    const url = `${baseUrl}${path}?${new URLSearchParams(params).toString()}`
+
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`)
+    }
+
+    // Validate the response
+    const json = await response.json()
+    if (isValidResponse) {
+      if (isValidResponse(json)) {
+        return json
+      } else {
+        throw new Error(
+          `Invalid response for Across API call ${path}. The response doesn't pass the validation. Did the API change?`
+        )
+      }
+    }
+
+    return response.json()
```

Comment:
By this point, `response.json()` has already been consumed.

```suggestion
    return json
```